doc/04-Upgrading.md

```diff
@@ -3,6 +3,64 @@
 Specific version upgrades are described below. Please note that version upgrades are incremental.
 If you are upgrading across multiple versions, make sure to follow the steps for each of them.
 
+## Upgrading to Icinga DB v1.1.2
+
+Please apply the `1.1.2.sql` upgrade script to your database. For package installations, you can find this file at
+`/usr/share/icingadb/schema/mysql/upgrades/` or `/usr/share/icingadb/schema/pgsql/upgrades/`, depending on your
+database vendor.
+
+!!! warning
+
+    In order to fix a loading performance issue of the history view in Icinga DB Web, this upgrade script adds an
+    appropriate index on the `history` table. Creating this new index normally takes place without blocking any other
+    queries. However, this may hang for a relatively considerable time, blocking all Icinga DB queries on all
+    `*_history` tables and the `history` table inclusively if there is an ongoing, long-running query on the `history`
+    table. One way of causing this to happen is if an Icinga Web user accesses the `icingadb/history` view at the exact
+    moment you are running this script. Depending on how many entries you have in the history table, Icinga DB Web may
+    take quite a long time to load, until your web servers timeout kicks in. 
+    See [unblock history tables](#unblock-history-tables) for details.
```

Comment:
> at the exact moment you are running this script

Not really. Given that it's a long-running query, it's actually a quite long time window. As you mention the web server timeout, I'm not sure about the details. At least when using the `icinga/icingaweb2` docker image, there seems to be no timeout or a very long timeout as you can see from [my `show processlist` output](https://github.com/Icinga/icingadb/pull/686#issuecomment-1988441677) where I had some of these queries running for over 1200s (=20min).

If there was a timeout that would abort the request and its query after say 30 seconds, I wouldn't really bother writing this in the documentation because the wait would be shorter than you'd need to even figure out that you're in this situation.

Reply:
I don't know how and why the configurations are getting lost with `icinga/icingaweb2` since we don't use this image for our dev environment, but there is for sure a default PHP [`max_execution_time`](https://www.php.net/manual/en/info.configuration.php) of `30s`.
```bash
root@768029d1c1dc:/var/www/html# grep -Rne 'max_execution_time' /usr/local/etc/php/
/usr/local/etc/php/php.ini:407:max_execution_time = 30
/usr/local/etc/php/php.ini-development:407:max_execution_time = 30
/usr/local/etc/php/php.ini-production:409:max_execution_time = 30
```

You can also set the execution timeout on a [web server](https://httpd.apache.org/docs/2.4/de/mod/core.html#timeout) basis, but we don't use apache2 for our dev setup either. However, we do set the [`FastCGI`](https://nginx.org/en/docs/http/ngx_http_fastcgi_module.html#fastcgi_read_timeout) read timeout for Nginx, which I think is useless as PHP aborts the requests after `30` by itself and doesn't wait for the CGI timeout to kick in.
```bash
root@e41052c1f5f3:/# grep -Rne 'fastcgi_read_timeout' /etc/nginx
/etc/nginx/nginx.conf:27:    fastcgi_read_timeout 3600s;
```